apps/platform-app/src/app/[locale]/(profile)/_workflows/profile-evaluation.workflow.ts

```diff
@@ -0,0 +1,152 @@
+'use server'
+import { AIMessage } from '@langchain/core/messages'
+import { ChatGroq } from '@langchain/groq'
+import { END, START, StateGraph } from '@langchain/langgraph'
+import { ToolNode } from '@langchain/langgraph/prebuilt'
+
+import {
+  sendProfileApprovedEmail,
+  sendProfileRejectedEmail,
+} from '@/backend/mailing/mailing.service'
+import { updateProfileById } from '@/backend/profile/profile.service'
+import { getUserById } from '@/backend/user/user.service'
+import { type UserWithProfileAndGH } from '@/backend/user/user.types'
+import { getContextVariable, setContextVariable } from '@langchain/core/context'
+import { tool } from '@langchain/core/tools'
+import { Annotation, type LangGraphRunnableConfig } from '@langchain/langgraph'
+import { z } from 'zod'
+
+const StateAnnotation = Annotation.Root({
+  user: Annotation<UserWithProfileAndGH>(),
+  evaluation: Annotation<string>(),
+})
+
+const evaluationModel = new ChatGroq({
+  model: 'llama-3.1-70b-versatile',
+  temperature: 0.2,
+  streaming: false,
+})
+
+const executionModel = new ChatGroq({
+  model: 'llama3-groq-8b-8192-tool-use-preview',
+  temperature: 0.1,
+})
+
+const acceptProfileTool = tool(
+  async (_, config: LangGraphRunnableConfig) => {
+    const userId = config.configurable?.userId
+    const currentState =
+      getContextVariable<typeof StateAnnotation.State>('currentState')
+
+    if (currentState?.user) {
```

Comment:
**nit:** we should throw some errors if if is not entered, because now it can not work and we will not anything about it